packages/std/src/math/decimal.rs

```diff
@@ -53,6 +55,31 @@ impl Decimal {
     pub fn is_zero(&self) -> bool {
         self.0 == 0
     }
+
+    /// Returns the multiplicative inverse `1/d` for decimal `d`.
+    ///
+    /// If `d` is zero, none is returned.
+    pub fn inv(&self) -> Option<Decimal> {
+        if self.is_zero() {
+            None
+        } else {
+            // Let self be p/q with p = self.0 and q = DECIMAL_FRACTIONAL_SQUARED.
+            // Now we calculate the inverse a/b = q/p such that b = DECIMAL_FRACTIONAL_SQUARED.
```

Comment:
I don't understand the comment. This is just scaled 1 (`10^18`) divided by (already scaled) decimal (`self.0`), re-scaled.

Reply:
Right, this says the same. Just `DECIMAL_FRACTIONAL_SQUARED` is wrong. It should be `DECIMAL_FRACTIONAL`. But yeah, it works better on paper than in a comment :/